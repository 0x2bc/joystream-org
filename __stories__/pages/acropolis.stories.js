import React from 'react';
import { storiesOf } from '@storybook/react';

import AcropolisPage from '../../src/pages/acropolis';

storiesOf('Pages|Acropolis', module).add('default', () => <AcropolisPage />);
