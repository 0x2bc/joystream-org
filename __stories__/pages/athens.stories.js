import React from 'react';
import { storiesOf } from '@storybook/react';

import AthensPage from '../../src/pages/athens';

storiesOf('Pages|Athens', module).add('default', () => <AthensPage />);
