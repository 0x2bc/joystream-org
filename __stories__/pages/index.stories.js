import React from 'react';
import { storiesOf } from '@storybook/react';

import IndexPage from '../../src/pages/index';

storiesOf('Pages|Home', module).add('default', () => <IndexPage />);
