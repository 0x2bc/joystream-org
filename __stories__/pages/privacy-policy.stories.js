import React from 'react';
import { storiesOf } from '@storybook/react';

import PrivacyPolicyPage from '../../src/pages/privacy-policy';

storiesOf('Pages|Privacy Policy', module).add('default', () => <PrivacyPolicyPage />);
