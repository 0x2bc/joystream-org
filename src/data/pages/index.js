import { ReactComponent as ValidatorsImage } from '../../assets/svg/active-validators.svg';
import { ReactComponent as StorageImage } from '../../assets/svg/storage.svg';
import { ReactComponent as MemberImage } from '../../assets/svg/council-member.svg';
import { ReactComponent as ScreenerImage } from '../../assets/svg/screener.svg';
import { ReactComponent as MembershipCuratorImage } from '../../assets/svg/membership-curator.svg';
import { ReactComponent as ContentCuratorImage } from '../../assets/svg/content-curator.svg';
import { ReactComponent as ContentCreatorImage } from '../../assets/svg/content-creator.svg';
import { ReactComponent as BandwidthProviderImage } from '../../assets/svg/bandwidth-provider.svg';
import { ReactComponent as DiscoveryProviderImage } from '../../assets/svg/discovery-provider.svg';
import { ReactComponent as LiveStreamingProviderImage } from '../../assets/svg/live-streaming-provider.svg';
import { ReactComponent as BuilderImage } from '../../assets/svg/builder.svg';
import { ReactComponent as CommunicationModeratorImage } from '../../assets/svg/communication-moderator.svg';

const roles = {
  active: [
    {
      image: ValidatorsImage,
      title: 'Validator',
      to: '/roles#Validator',
      key: 'validatorsCount',
      hasLabel: false,
    },
    {
      image: MemberImage,
      title: 'Council Member',
      to: '/roles#Council-Member',
      key: 'councilMembersCount',
      hasLabel: false,
    },
    {
      image: StorageImage,
      title: 'Storage Provider',
      to: '/roles#Storage-Provider',
      key: 'storageProviders',
      hasLabel: true,
    },
  ],
  future: [
    { image: ScreenerImage, title: 'Membership Screener', to: '/roles' },
    {
      image: MembershipCuratorImage,
      title: 'Membership Curator',
      to: '/roles',
    },
    { image: ContentCuratorImage, title: 'Content Curator', to: '/roles' },
    { image: ContentCreatorImage, title: 'Content Creator', to: '/roles' },
    {
      image: BandwidthProviderImage,
      title: 'Bandwidth Provider',
      to: '/roles',
    },
    {
      image: DiscoveryProviderImage,
      title: 'Discovery Provider',
      to: '/roles',
    },
    {
      image: LiveStreamingProviderImage,
      title: 'Live Streaming Provider',
      to: '/roles',
    },
    { image: BuilderImage, title: 'Builder', to: '/roles' },
    {
      image: CommunicationModeratorImage,
      title: 'Communication Moderator',
      to: '/roles',
    },
  ],
};

export { roles };
