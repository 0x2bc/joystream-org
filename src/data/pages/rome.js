const goalsData = [
  {
    title: 'rome.testnetGoals.goals.contentDirectory.title',
    text: 'rome.testnetGoals.goals.contentDirectory.text',

  },
  {
    title: 'rome.testnetGoals.goals.content.title',
    text: 'rome.testnetGoals.goals.content.text',

  },
  {
    title: 'rome.testnetGoals.goals.contentCurator.title',
    text: 'rome.testnetGoals.goals.contentCurator.text',

  },
  {
    title: 'rome.testnetGoals.goals.contentCreator.title',
    text: 'rome.testnetGoals.goals.contentCreator.text',
    
  },
];

const launchDate = '2020/03/17 15:00';

export { goalsData, launchDate };
