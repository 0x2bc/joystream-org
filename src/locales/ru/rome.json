{
  "rome": {
    "siteMetadata": {
      "description": "Изучаем тестнет Rome"
    },
    "hero": {
      "title": "Сеть Rome",
      "text": "Изучайте доступные роли и выбирайте наиболее подходящие вам. Влияйте на развитие платформы, зарабатывая при этом реальные деньги.",
      "chipText": "Что это?"
    },
    "heroCard": {
      "title": "ЗАМЕНЕН НА CONSTANTINOPLE"
    },
    "modal": {
      "title": "Rome",
      "text": "<0>Римская империя оставила богатое наследие.</0> Во время ее расцвета архитектурные и инженерные навыки римлян не имели себе равных. Хотя идея акведуков не была изобретена римлянами, их красота и протяженность поражают."
    },
    "criticalDocuments": {
      "title": "Важная документация",
      "fullSpecifications": {
        "title": "Полное описание",
        "text": "Характеристики тестнета Rome не были опубликованы."
      },
      "releasePlan": {
        "title": "План релиза",
        "text": "Прочтите план релиза в том виде, в котором он был составлен на этапе планирования и сравните с тем, как прошла разработка."
      }
    },
    "testnetGoals": {
      "title": "Цели тестнета",
      "subtitle": "Указанные ниже цели дают общее представление о ключевых результатах, перечисленных в нашем релизе <0>OKR</0>.",
      "goals": {
        "contentDirectory": {
          "title": "Создание динамичного, гибкого и удобного каталога содержимого",
          "text": "Краеугольным камнем тестнета Rome является введение нового каталога контента, который можно расширять, изменять и улучшать без необходимости обновления среды выполнения. Им и обусловлены все основные цели и результаты."
        },
        "content": {
          "title": "Заполнение каталога различным контентом",
          "text": "Мы выпускаем в Rome новую систему Каталога контента, которая, с одной стороны, рассчитанна на длительный срок службы, а с другой - на постоянные улучшения. Мы будем оплачивать загрузку различного содержимого Создателям контента, надеясь, что результаты нас многому научат."
        },
        "contentCurator": {
          "title": "Введение новой роли Контент куратора",
          "text": "С введением нового каталога контента и поощрений за публикации, нам необходим мониторинг. Чтобы стать Контент куратором, необходимо будет выполнить процедуру стейкинга. Кураторы будут получать оплату как в токенах тестнета, так и в Monero."
        },
        "contentCreator": {
          "title": "Добавление профилей Создателей контента",
          "text": "Без Создателей контента не было бы никакого контента. Может показаться, что кроме необходимости быть участником, добавление новой роли выглядит как дополнительное препятствие, но это позволит пользователям загружать контент, используя различные профили."
        }
      }
    },
    "roles": {
      "title": "Поощряемые роли в сети Rome"
    },
    "map": {
      "title": "Rome",
      "text": "<0>Рим был столицей великой Римской империи в период ее расцвета.</0> Как и в случае с предыдущими названиями тестнетов, Римская империя и Рим были еще одним важным шагом в развитии демократии, верховенства закона и современных структур управления.<1/><1/><2><0/> Информация о тестнете</2>"
    }
  }
}
