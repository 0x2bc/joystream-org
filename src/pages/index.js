import React from 'react';
import { pagePropTypes } from '../propTypes';

import getApiPath from '../utils/getApiPath';
import mapStatusDataToAnalytics from '../utils/mapStatusDataToAnalytics';
import mapStatusDataToRoles from '../utils/mapStatusDataToRoles';

import withApi from '../components/_enhancers/withApi';

import BaseLayout from '../components/_layouts/Base';
import Link from '../components/Link';
import Subheader from '../components/Subheader';
import TestnetItem from '../components/TestnetItem';
import Analytics from '../components/Analytics';
import Button from '../components/Button';
import TitleWrapper from '../components/TitleWrapper';
import RoleList from '../components/RoleList';
import ColumnsLayout from '../components/ColumnsLayout';
import LayoutWrapper from '../components/LayoutWrapper';
import Hero from '../components/Hero';
import SiteMetadata from '../components/SiteMetadata';

import AcropolisImage from '../assets/svg/acropolis-main.svg';
import heroImage from '../assets/svg/hero-builder.svg';
import RomeImage from '../assets/svg/rome-main.svg';
import constantinopleImage from '../assets/svg/constantinople-main.svg';
import alexandriaImage from '../assets/svg/alexandria-main.svg';
import { ReactComponent as TickImage } from '../assets/svg/tick.svg';
import { ReactComponent as ClockImage } from '../assets/svg/clock.svg';

import { roles } from '../data/pages';
import { launchDate as alexandriaNetworkLaunchDate } from '../data/pages/alexandria';

import './style.scss';

const activeTestnet = {
  name: 'Alexandria',
  incentivesLink: 'https://www.joystream.org/testnet',
};

const IndexPage = ({ content }) => (
  <BaseLayout>
    <SiteMetadata
      title="Joystream: The video platform DAO"
      description="Joystream is a video platform controlled, owned, and operated by its users."
    />
    <Hero
      image={heroImage}
      title={
        <>
          The video <br /> platform DAO
        </>
      }
      animationStartValue={0}
    >
      <p className="IndexPage__hero-paragraph">
        Joystream is a video platform controlled, owned, and operated by its users.
      </p>
      <div className="IndexPage__hero-group">
        <Button noWrap large secondary className="IndexPage__hero-button" href="/testnet/">
          Earn Tokens
        </Button>
        <Button
          noWrap
          large
          secondary
          reversed
          className="IndexPage__hero-button"
          href="https://testnet.joystream.org/"
        >
          Launch UI
        </Button>
      </div>
    </Hero>

    <LayoutWrapper>
      <TestnetItem
        title="Alexandria Testnet"
        image={alexandriaImage}
        children={
          <>Alexandria is our seventh testnet, introducing a variety of important technical improvements.</>
        }
        button={{
          label: 'Explore Alexandria',
          to: '/alexandria',
        }}
      />
    </LayoutWrapper>

    <Analytics title="Testnet Metrics" large content={mapStatusDataToAnalytics(content)}>
      <Button secondary href={activeTestnet.incentivesLink}>
        Participate and Earn Monero
      </Button>
    </Analytics>

    {/*

    <LayoutWrapper>
      <TestnetItem
        title="Alexandria Testnet"
        image={alexandriaImage}
        children="Alexandria is our seventh testnet, focused on a variety of important technical improvements."
        date={alexandriaNetworkLaunchDate}
        button={{
          label: 'Explore Alexandria',
          to: '/alexandria',
        }}
      />
    </LayoutWrapper>

    */}

    <LayoutWrapper dark>
      <TitleWrapper title="Become a user, run the platform">
        <Subheader
          title="Active roles on the current testnet"
          content="learn more, join a role and subscribe for more"
          icon={TickImage}
        />
        <ColumnsLayout>
          <RoleList roles={roles.active} content={mapStatusDataToRoles(content)} />
        </ColumnsLayout>
        <Subheader
          title="Roles coming in future testnets"
          content="choose a role, learn more and subscribe to get updated"
          icon={ClockImage}
        />
        <ColumnsLayout columnsCount={3}>
          <RoleList roles={roles.future} content={mapStatusDataToRoles(content)} />
        </ColumnsLayout>

        <Button secondary to="/roles" className="IndexPage__roles-button">
          Learn more about the roles
        </Button>
      </TitleWrapper>
    </LayoutWrapper>
  </BaseLayout>
);

IndexPage.propTypes = pagePropTypes;

export { IndexPage };
export default withApi(IndexPage, getApiPath('/'));
