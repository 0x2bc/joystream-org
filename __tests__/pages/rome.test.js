import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { RomePage } from '../../src/pages/rome';
import testContent from '../../__mocks__/data/testContent';

configure({ adapter: new Adapter() });

describe('RomePage page', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RomePage content={testContent} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
