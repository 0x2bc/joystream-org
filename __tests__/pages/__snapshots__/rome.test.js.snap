// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RomePage page renders correctly 1`] = `
<BaseLayout>
  <SiteMetadata
    description="rome.siteMetadata.description"
    image="test-file-stub"
    lang="en"
    meta={Array []}
    title="siteMetadata.title"
  />
  <Hero
    animationEnd={500}
    animationEndValue={-40}
    animationStartValue={0}
    chip={
      <Chip
        icon="icon-component-mock"
        onClick={[Function]}
      >
        rome.hero.chipText
      </Chip>
    }
    image="icon-mock"
    indent={true}
    noOverflow={false}
    reverse={false}
    theme="black"
    title="rome.hero.title"
  >
    <p
      className="RomePage__hero-paragraph"
    >
      rome.hero.text
    </p>
    <HeroCard
      content={null}
      counterTitle={
        <React.Fragment>
          rome.heroCard.title
        </React.Fragment>
      }
      date="2020/05/20 15:00"
      error={false}
      info={true}
    />
    <TestnetModal
      className=""
      closeModal={[Function]}
      image="icon-mock"
      isOpen={false}
      title="rome.modal.title"
    >
      <p>
        <Trans
          components={
            Array [
              <strong />,
            ]
          }
          i18nKey="rome.modal.text"
        />
      </p>
    </TestnetModal>
  </Hero>
  <LayoutWrapper
    className=""
    dark={false}
    gradient={false}
  >
    <TitleWrapper
      className=""
      subtitle=""
      title="rome.criticalDocuments.title"
    >
      <ColumnsLayout
        className=""
        columnsCount={2}
        largeSpacing={false}
      >
        <Pane
          className=""
          disabled={true}
          image="icon-component-mock"
          title="rome.criticalDocuments.fullSpecifications.title"
        >
          rome.criticalDocuments.fullSpecifications.text
        </Pane>
        <Pane
          className=""
          disabled={false}
          href="https://github.com/Joystream/joystream-landing/tree/master/testnets/rome"
          image="icon-component-mock"
          target="_blank"
          title="rome.criticalDocuments.releasePlan.title"
        >
          rome.criticalDocuments.releasePlan.text
        </Pane>
      </ColumnsLayout>
    </TitleWrapper>
    <TitleWrapper
      className=""
      subtitle={
        <React.Fragment>
          <Trans
            components={
              Array [
                <Link
                  className=""
                  highlighted={false}
                  href="https://github.com/Joystream/joystream-landing/tree/master/testnets/rome#release-okrs"
                >
                  OKR
                </Link>,
              ]
            }
            i18nKey="rome.testnetGoals.subtitle"
          />
        </React.Fragment>
      }
      title="rome.testnetGoals.title"
    >
      <GoalList
        data={
          Array [
            Object {
              "text": "rome.testnetGoals.goals.contentDirectory.text",
              "title": "rome.testnetGoals.goals.contentDirectory.title",
            },
            Object {
              "text": "rome.testnetGoals.goals.content.text",
              "title": "rome.testnetGoals.goals.content.title",
            },
            Object {
              "text": "rome.testnetGoals.goals.contentCurator.text",
              "title": "rome.testnetGoals.goals.contentCurator.title",
            },
            Object {
              "text": "rome.testnetGoals.goals.contentCreator.text",
              "title": "rome.testnetGoals.goals.contentCreator.title",
            },
          ]
        }
      />
    </TitleWrapper>
  </LayoutWrapper>
  <LayoutWrapper
    className=""
    dark={true}
    gradient={false}
  >
    <TitleWrapper
      className=""
      subtitle=""
      title="rome.roles.title"
    >
      <ColumnsLayout
        className=""
        columnsCount={2}
        largeSpacing={false}
      >
        <RoleList
          content={
            Object {
              "contentCreatorsCount": 20,
              "contentCuratorsCount": 3,
              "councilMembersCount": 12,
              "storageProviders": 10,
              "validatorsCount": 20,
            }
          }
          oldTestnet={true}
          roles={
            Array [
              Object {
                "hasLabel": false,
                "image": "icon-component-mock",
                "key": "contentCreatorsCount",
                "title": "rolesData.contentCreator",
                "to": "/roles#content-creator",
              },
              Object {
                "hasLabel": false,
                "image": "icon-component-mock",
                "key": "contentCuratorsCount",
                "title": "rolesData.contentCurator",
                "to": "/roles#content-curator",
              },
              Object {
                "hasLabel": false,
                "image": "icon-component-mock",
                "key": "validatorsCount",
                "title": "rolesData.validator",
                "to": "/roles#validator",
              },
              Object {
                "hasLabel": false,
                "image": "icon-component-mock",
                "key": "councilMembersCount",
                "title": "rolesData.councilMember",
                "to": "/roles#council-member",
              },
              Object {
                "hasLabel": false,
                "image": "icon-component-mock",
                "key": "storageProviders",
                "title": "rolesData.storageProvider",
                "to": "/roles#storage-provider",
              },
            ]
          }
        />
      </ColumnsLayout>
    </TitleWrapper>
  </LayoutWrapper>
  <MapInfo
    className=""
    location="rome"
    title="rome.map.title"
  >
    <p>
      <Trans
        components={
          Array [
            <strong />,
            <br />,
            <icon-component-mock />,
            <Link
              className=""
              highlighted={false}
              href="https://testnet.joystream.org/"
            >
              Explore current testnet
            </Link>,
          ]
        }
        i18nKey="rome.map.text"
      />
    </p>
  </MapInfo>
</BaseLayout>
`;
